javascript/javascript2/week3/script.js

```diff
@@ -0,0 +1,128 @@
+// 1. Logout text
+let logOut = () => {
+  console.log('Script loaded');
+};
+setTimeout(logOut, 2500);
+
+// 2. Delay function
+
+function setTime(n, callback, time = 0) {
+  setTimeout(() => {
+    console.log(`loading after ${(n + time) / 1000} second `);
+    if (callback) {
+      callback();
+    }
+  }, n);
+}
+setTime(2000, () => {
+  setTime(3000, null, 2000);
+});
+
+// 3. Button task
+let button = document.getElementById('clickBtn');
+button.addEventListener('click', () => {
+  setTimeout(() => {
+    console.log('Called after 5 second');
+  }, 5000);
+});
+
+// 4. Planet log function (TEST)
+function earthLogger() {
+  console.log('Earth');
+}
+function saturnLogger() {
+  console.log('Saturn');
+}
+function planetLogFunction(callback) {
+  callback();
+}
+planetLogFunction(earthLogger);
+planetLogFunction(saturnLogger);
+
+// 5. Log Location
+let button2 = document.getElementById('clickBtn2');
+button2.addEventListener('click', () => {
+  navigator.geolocation.getCurrentPosition((position) => {
+    console.log(
+      `Latitude :${position.coords.latitude} Longitude:${position.coords.longitude}`
+    );
+  });
+});
+
+// 6. Function run after delay
+function runAfterDelay(n, callback) {
+  setTimeout(() => {
+    console.log(`Should be logged after ${n} second `);
+    if (callback) {
+      callback();
+    }
+  }, n * 1000);
+}
+runAfterDelay(3, null);
+
+// 7. Double click
+let myBtn = document.getElementById('clickBtn');
+myBtn.addEventListener('dblclick', () => {
+  setTimeout(() => {
+    console.log('Its double click');
+  }, 500);
+});
+
```

Comment:
I don't think the set timeout here is actually having anything for the dblclick, you can change the number and see that it doesn't change it. Instead the only way I can think of is with an if condition and a set timeout. But if you have any documentation that dblclick indeed works this way, I will be more than happy to read it!

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Element/dblclick_event this is the documentation I found about double click 🙂